src/main/java/org/broadinstitute/hellbender/engine/ParsedArgument.java

```diff
@@ -0,0 +1,142 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Represents a parsed argument for the FeatureInput and ReadInput.
+ * Always has a file and a name.
+ * May have attributes: key-value mappings.
+ *
+ * ParsedArgument can be assigned logical names on the command line using the syntax:
+ *
+ *     --argument_name logical_name:feature_file
+ *
+ * These logical names can then be retrieved by the tool at runtime via {@link #getName}
+ *
+ * Furthermore, a list of comma-separated key=value pairs may be provided as follows:
+ *
+ *     --argument_name logical_name,key1=value1,key2=value2:feature_file
+ *
+ * the string value provided for a given key can be retrieved via {@link #getAttribute(String)}. Keys must be unique.
+ */
+final class ParsedArgument {
```

Comment:
Also, I think the name of this class is too generic. Recommend renaming to `TaggedArgument`